src/ext/response-targets.js

```diff
@@ -3,43 +3,55 @@
     /** @type {import("../htmx").HtmxInternalApi} */
     var api;
 
-    const targetAttrPrefix = 'hx-target-';
-    const targetAttrMinLen = targetAttrPrefix.length - 1;
+    const attrPrefix = 'hx-target-';
 
     /**
      * @param {HTMLElement} elt
      * @param {number} respCode
      * @returns {HTMLElement | null}
      */
-    function getRespCodeTarget(elt, respCode) {
-        if (!elt || !respCode) return null;
+    function getRespCodeTarget(elt, respCodeNumber) {
+        if (!elt || !respCodeNumber) return null;
 
-        var targetAttr = targetAttrPrefix + respCode;
-        var targetStr  = api.getClosestAttributeValue(elt, targetAttr);
+        var respCode = respCodeNumber.toString();
 
-        if (targetStr) {
-            if (targetStr === "this") {
-                return api.findThisElement(elt, targetAttr);
-            } else {
-                return api.querySelectorExt(elt, targetStr);
-            }
-        } else {
-            for (let l = targetAttr.length - 1; l > targetAttrMinLen; l--) {
-                targetAttr = targetAttr.substring(0, l) + '*';
```

Comment:
I like the explicitness of your solution, but I'm also curious why you didn't just add:

```
targetAttrStar = targetAttr.substring(0, l) + '*'
targetAttrX = targetAttr.substring(0, l) + 'x'
targetStr  = api.getClosestAttributeValue(elt, targetAttrStar) || api.getClosestAttributeValue(elt, targetAttrX)

Reply:
> I like the explicitness of your solution, but I'm also curious why you didn't just add:

I liked keeping all the possibilities in one place, where they could be understood separately from understanding how the attribute possibilities get processed. Makes it easy for someone to check what's supported.

I don't mind changing it if you want.

Reply:
Rebased!